addons/pos_self_order/static/src/kiosk/pages/product/product.js

```diff
@@ -0,0 +1,149 @@
+/** @odoo-module */
+
+import { Component, onWillUnmount, useState } from "@odoo/owl";
+import { useselfOrder } from "@pos_self_order/kiosk/self_order_kiosk_service";
+import { AttributeSelection } from "@pos_self_order/kiosk/components/attribute_selection/attribute_selection";
+import { KioskTemplate } from "@pos_self_order/kiosk/template/kiosk_template";
+import { Line } from "@pos_self_order/common/models/line";
+import { useService } from "@web/core/utils/hooks";
+
+export class Product extends Component {
+    static template = "pos_self_order.Product";
+    static props = ["product", "back?", "onValidate?"];
+    static components = { AttributeSelection, KioskTemplate };
+
+    setup() {
+        this.selfOrder = useselfOrder();
+        this.selfOrder.lastEditedProductId = this.props.product.id;
+        this.router = useService("router");
+
+        if (!this.props.product) {
+            this.router.navigate("productList");
+            return;
+        }
```

Comment:
This verification should be done before the lines using it (thus before line 17)